### PR TITLE
Add AREASCAL in the list of columns read from OGIP PHA file and pass …

### PR DESCRIPTION
**Pull request by registerrier**
…it to the PHACountSpectrum constructor.

This should solve the issue https://github.com/gammasky/joint-crab/issues/48 @cdeil and @cosimoNigro

### CODE REVIEW

**Comment by cdeil** [comment]
This is breaking a few existing tests:
https://travis-ci.org/gammapy/gammapy/jobs/347870845#L3766

@registerrier @joleroi Is raising a KeyError on those files appropriate? Or are they valid cases and there's e.g. a default value that could / should be assumed?

**Comment by registerrier** [comment]
Right I missed this test. The issue is that the AREASCAL column does not exist for the spectra stored on GAMMAPY_EXTRA. 
There should be a test or a try/except on the presence of columns in the read method

**Comment by joleroi** [comment]
@registerrier @cosimoNigro @cdeil Thanks a lot for cleaning up my mess :grin: 

> Right I missed this test. The issue is that the AREASCAL column does not exist for the spectra stored on GAMMAPY_EXTRA.

I'd rather update ``gammapy-extra``. The AREASCAL column should always be present and maybe filled with 1. Do you agree @registerrier ? I can do this change.

**Comment by registerrier** [comment]
There might be cases where the AREASCAL is not there, e.g. if it has been produced by other means or specific scripts made by some people. In the long run we might want to do MWL fits with X-ray spectra which won't have the column in most of the cases. 
I would say it is better to have the possibility to read it and possibly print a warning for more safety.

**Comment by registerrier** [comment]
BTW, in `PHACountSpectrum.to_sherpa` areascal is hardcoded at 1. Is it the expected behavior or should it be:
```
return DataPHA(
            name=name,
            channel=(table['CHANNEL'].data + 1).astype(SherpaFloat),
            counts=table['COUNTS'].data.astype(SherpaFloat),
            quality=table['QUALITY'].data,
            exposure=self.livetime.to('s').value,
            backscal=backscal,
            areascal=self.areascal,
            syserror=None,
            staterror=None,
            grouping=None,
        )```

**Comment by joleroi** [comment]
> BTW, in PHACountSpectrum.to_sherpa areascal is hardcoded at 1. Is it the expected behavior or should it be:

To me, this seems like a :beetle:

**Comment by registerrier** [comment]
OK. 
It seems many tests are broken now. In particular, those testing the fit results.
I assume they were reading from the existing ogip files. How should we proceed?
Replace the existing value in the tests I suppose.

In many cases the relative tolerance of 1e-7 is too strong. 
How much would be a reasonable value?
 e.g. for spectral indices: is an absolute tolerance of 0.01-0.02 not sufficient?

**Comment by joleroi** [comment]
@registerrier 

The files in gammapy-extra don't have an ``AREASCAL`` column, so I'm surprised that the fit results change. Note that this is not due to the tolerance, e.g. [here](https://travis-ci.org/gammapy/gammapy/jobs/348175227#L2028)

I am looking in to this now

File: gammapy/spectrum/core.py [hunk 1]
```diff
@@ -473,12 +473,24 @@ def from_hdulist(cls, hdulist, hdu1='SPECTRUM', hdu2='EBOUNDS'):
         ebounds = fits_table_to_table(hdulist[2])
         emin = ebounds['E_MIN'].quantity
         emax = ebounds['E_MAX'].quantity
+
+        # Check if column are present in the header
+        quality=None
+        areascal=None
+        backscal=None
+        if 'QUALITY' in counts_table:
```
Comment (joleroi) [inline gammapy/spectrum/core.py]: This should be 

```
if 'QUALITY' in counts_table.colnames:
```

This is also the reasons why the tests fail ...
@registerrier I tried to push the changes to your branch but apparently I don't have access. Either you grant acess to me or you correct this yourself, up to you :smile:

**Comment by registerrier** [comment]
Ok thanks for the correction @joleroi. Sorry for that. I have committed the change now all tests run.

**Comment by joleroi** [comment]
@cdeil please approve, I don't like hitting a red :record_button:

**Comment by cdeil** [comment]
@registerrier @joleroi - Thanks!

@cosimoNigro - Please update to Gammapy master and update the analyses in the joint Crab repo.

@lmohrmann - If you use OGIP spec I/O from Gammapy for the HESS checks, you might want to update to latest Gammapy and re-run.

**Comment by cdeil** [comment]
I re-ran the spectral analyses in https://github.com/gammasky/HESS-DL3-DR1/commit/b8c26ab3c292e7879b344120789c7bcadaeeac40 via 
```
time ./make.py analysis-spec-gammapy --extract --fit
```
The fit results are identical.

There I do store OGIP files, read them back and then do the fit.
@joleroi - do you understand why the analysis results there aren't affected?

**Pull request merged**